src/Statement.py

```diff
@@ -24,45 +24,54 @@ def get_category(self):
 
-        :return: category of the statement; if unknown return the empty string
+        :return: category of the statement; if ignored on purpose return 'Ignored', if unknown return the empty string
         """
         booking_type = self._statement[self._config.get_booking_type()]
-        value = self.get_value()
         category = ""
-        if self._config.get_relevant_income_regex().match(booking_type):
-            category = "Zinsen"
-        elif self._config.get_relevant_invest_regex().match(booking_type):
-            category = "Einlage"
-        elif self._config.get_relevant_payment_regex().match(booking_type):
-            category = "Entnahme"
-        elif self.is_fee(booking_type):
-            category = "Gebühren"
-        elif self._config.get_special_entry_regex() and self._config.get_special_entry_regex().match(booking_type):
-            if value > 0:
-                category = "Zinsen"
-            elif value < 0:
-                category = "Gebühren"
-            else:
-                logging.debug(self._statement)
-        elif self._config.get_ignorable_entry_regex() and self._config.get_ignorable_entry_regex().match(booking_type):
-            category = "Ignored"
-        else:
-            logging.debug(self._statement)
+
+        regex_to_category_mappings = [
+            {"regex": self._config.get_relevant_income_regex(), "category": "Zinsen"},
+            {"regex": self._config.get_relevant_invest_regex(), "category": "Einlage"},
+            {"regex": self._config.get_relevant_payment_regex(), "category": "Entnahme"},
+            {"regex": self._config.get_relevant_fee_regex(), "category": "Gebühren"},
+            {"regex": self._config.get_special_entry_regex(), "category": "Undecided"},
+            {"regex": self._config.get_ignorable_entry_regex(), "category": "Ignored"},
+        ]
+
+        for mapping in regex_to_category_mappings:
+            if mapping["regex"] and mapping["regex"].match(booking_type):
+                category = mapping["category"]
+                break
+
+        if category == "Undecided":
+            category = Statement.__handle_special_case_mintos_discount_premium(self.get_value())
+
+        if category == "":
+            logging.debug("Unexpected statement: ", self._statement)
+
         return category
 
-    def is_fee(self, booking_type):
-        """ check if it is a statement with a fee"""
-        return self._config.get_relevant_fee_regex() and self._config.get_relevant_fee_regex().match(booking_type)
+    @staticmethod
+    def __handle_special_case_mintos_discount_premium(value):
+        # This is currently a special case for the Mintos "discount/premium" secondary market transactions parsing,
+        # where an entry might be a fee or an income depending on its sign.
+
+        if value >= 0:
+            return "Zinsen"
+        elif value < 0:
+            return "Gebühren"
+        else:
+            return "Ignored"
```

Comment:
Please correct me if I'm wrong, the else statement is only for the case where value is not a number, right? hm. Should work I guess.

Reply:
Yes exactly, it might be unexpected, but at least that way the "Undecided" will not make it into the next higher level where it is not handled.